src/lets_be_rational/intrinsic.rs

```diff
@@ -0,0 +1,135 @@
+use crate::OptionType;
+
+const FOURTH_ROOT_DBL_EPSILON: f64 = 0.0001220703125;
+
+const NORMALISED_X2_THRESHOLD: f64 = 98.0 * FOURTH_ROOT_DBL_EPSILON;
+
+pub(crate) fn normalised_intrinsic(x: f64, option_type: OptionType) -> f64 {
+    let q = option_type as i32 as f64;
+    if q * x <= 0.0 {
+        return 0.0;
+    }
+    let x2 = x * x;
+    if x2 < NORMALISED_X2_THRESHOLD {
+        return ((if q < 0.0 { -1.0 } else { 1.0 })
+            * x
+            * (1.0
+                + x2 * ((1.0 / 24.0)
+                    + x2 * ((1.0 / 1920.0) + x2 * ((1.0 / 322560.0) + (1.0 / 92897280.0) * x2)))))
+            .max(0.0)
+            .abs();
+    }
+    let b_max = (0.5 * x).exp();
+    let one_over_b_max = 1.0 / b_max;
+    ((if q < 0.0 { -1.0 } else { 1.0 }) * (b_max - one_over_b_max))
+        .max(0.0)
+        .abs()
+}
+
+#[cfg(test)]
+mod tests {
+    use approx::assert_relative_eq;
+    use proptest::prelude::*;
+
+    use super::*;
+
+    const EPSILON: f64 = 1e-9;
+
+    #[test]
+    fn test_normalised_intrinsic_near_zero() {
+        // arrange
+        let x = 0.0001;
+        const EXPECTED_RESULT: f64 = 0.00010000000004166668;
+        let option_type = OptionType::Call;
+
+        // act
+        let result = normalised_intrinsic(x, option_type);
+
+        // assert
+        assert_relative_eq!(result, EXPECTED_RESULT, epsilon = EPSILON);
+    }
+
+    #[test]
+    fn test_normalised_intrinsic_with_large_x() {
+        let x = f64::MAX;
+        let option_type = OptionType::Call;
+        let result = normalised_intrinsic(x, option_type);
+
+        // TODO: to analyze correct paths here
```

Comment:
for f64:MAX we have inf+. in my opinion we shouldn't get inf. I think we should consider some of error of different behaviour on another level. Or we should return Option